File: solutions/erlang/rna-transcription/1/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).


to_rna([]) -> [];
to_rna([Head|Tail]) -> 
    [to_rna(Head) | to_rna(Tail)];
to_rna($C) -> $G;
to_rna($G) -> $C;
to_rna($T) -> $A;
to_rna($A) -> $U.

File: solutions/cpp/rna-transcription/1/rna_transcription.cpp
#include "rna_transcription.h"

#include <algorithm>
#include <functional>

namespace rna_transcription {
  char to_rna(const char& nucleotid) {
    switch (nucleotid) {
      case 'G': return 'C'; break;
      case 'C': return 'G'; break;
      case 'A': return 'U'; break;
      case 'T': return 'A'; break;
      default: return ' ';
    }
  }

  std::string to_rna(const std::string& dna) {
    std::string rna(dna.size(), ' ');
    std::transform(
        begin(dna), end(dna), begin(rna), std::ref<char(const char&)>(to_rna));
    return rna;
  }
}  // namespace rna_transcription
